docs/developer-guide.md

```diff
@@ -0,0 +1,139 @@
+
+_Keep in mind `cri-o` can also be used in place of `containerd/cri`, but for this guide we will be using `containerd/cri`. You may need some additional steps to ensure `cri-o` works, but the setup should be mostly the same. You can also follow the `cri-o` [install guide](https://github.com/cri-o/cri-o/blob/master/install.md#install-packaged-versions-of-cri-o) if you prefer to use the packaged version._
+
+- Clone / Fork the `cri` repo from [https://github.com/containerd/cri](https://github.com/containerd/cri)
+
+- Pulldown the forked version of `cri` into your `$HOME/go/src/github.com/containerd/cri` and set the `upstream` repo to original project repo url.
+  - Note, as mentioned before that the folder you save the repo (`cri`) to locally must follow the same structure as what you see for the repositories path on `github.com`
+- `cd` into your newly cloned `cri` repo on your machine
+
+  ```bash
+    $ cd $HOME/go/src/github.com/containerd/cri
+    $
+  ```
+
+_If you are not running as `root` (which you shouldn't) the commands below may fail on the first try, re-run them by prefixing `sudo` in front to ensure any issues you encounter are not permissions related; example `sudo <command>`_
+
+- Install any necessary build tools using `make`
+
+  ```bash
+    $ make install.tools
+  ```
+
+- Install any necessary dependencies using `make`
+
+  ```bash
+    $ make install.deps
+  ```
+
+- Build `containerd` from source and install its binary using `make`
+
+  ```bash
+    $ service containerd stop # Stop containerd first, if already running
+    $ make && make install
+  ```
+
+- Run `containerd -v`  to validate you are using the development version of `containerd`
+
+  ```bash
+    $ containerd -v
+    WARN[2020-09-18T21:29:27.616910535Z] This customized containerd is only for CI test, DO NOT use it for distribution.
+    containerd github.com/containerd/containerd a536d06c-TEST
+  ```
+
+- Update the `containerd` config by copying the output from the `containerd config default` command and paste this output to your `containerd` config file which should be located under `/etc/containerd/config.toml`
+
+  ```bash
+    $ containerd config default # Remaining output truncated -- copy the output from this command
+    WARN[2020-09-18T21:37:38.987970869Z] This customized containerd is only for CI test, DO NOT use it for distribution.
+    version = 2
+    root = "/var/lib/containerd"
+    state = "/run/containerd"
+    ...
+  ```
+
+  _Background for why this is necessary: `docker` makes use of `containerd`, but when installed it has `cri` disabled, the updated config enables `cri` and allows `docker` to make use of the development `containerd/cri` that you have been installing up to this point_
```

Comment:
I think this is very valuable information on using containerd/cri. I feel however that it should reside in [containerd/cri dev guide](https://github.com/containerd/cri#getting-started-for-developers). Do you mind opening a PR there and updating its guide where necessary.

I feel in here it should just contain:

Follow the [containerd/cri dev guide](https://github.com/containerd/cri#getting-started-for-developers) to make and install containerd/cri runtime.